src/Microsoft.AspNet.OData/Formatter/Deserialization/ODataDeserializerContext.cs

```diff
@@ -27,6 +28,12 @@ public HttpRequestMessage Request
                 _request = value;
                 InternalRequest = _request != null ? new WebApiRequestMessage(_request) : null;
                 InternalUrlHelper = _request != null ? new WebApiUrlHelper(_request.GetUrlHelper()) : null;
+
+                var webApiRequestMessage = (WebApiRequestMessage)InternalRequest;
```

Comment:
>var [](start = 16, length = 3)

We typically try and avoid "var" in our production code as it makes it a bit harder to understand what is intended.

Reply:
#Resolved